doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -384,433 +381,602 @@ Rewriting with definitional equality
+   recursive calls as well as those designated by :cmd:`Arguments` :n:`@reference … /`
+   commands. For instance, a constant :g:`plus' := plus` may be unfolded and
+   reused in recursive calls, but a constant such as :g:`succ := plus (S O)` is
+   not unfolded unless it was specifically designated in an :cmd:`Arguments`
+   command such as :n:`Arguments succ /.`.
 
-      This flag causes all calls to the native compiler to print
-      timing information for the conversion to native code,
-      compilation, execution, and reification phases of native
-      compilation.  Timing is printed in units of seconds of
-      wall-clock time.
+   :n:`{| @reference_occs | @pattern_occs }` can limit the application of :tacn:`simpl`
+   to:
 
-   .. flag:: NativeCompute Profiling
+   - applicative subterms whose :term:`head` is the
+     constant :n:`@qualid` or is the constant used
+     in the notation :n:`@string` (see :n:`@reference`)
+   - subterms matching a pattern :n:`@one_term`
 
-      On Linux, if you have the ``perf`` profiler installed, this flag makes
-      it possible to profile :tacn:`native_compute` evaluations.
+.. tacn:: cbn {? @reductions } @simple_occurrences
 
-   .. opt:: NativeCompute Profile Filename @string
-      :name: NativeCompute Profile Filename
+   :tacn:`cbn` was intended to be a more principled, faster and more
+   predictable replacement for :tacn:`simpl`.
 
-      This option specifies the profile output; the default is
-      ``native_compute_profile.data``. The actual filename used
-      will contain extra characters to avoid overwriting an existing file; that
-      filename is reported to the user.
-      That means you can individually profile multiple uses of
-      :tacn:`native_compute` in a script. From the Linux command line, run ``perf report``
-      on the profile file to see the results. Consult the ``perf`` documentation
-      for more details.
+   The main difference between :tacn:`cbn` and :tacn:`simpl` is that
+   :tacn:`cbn` may unfold constants even when they cannot be reused in recursive calls:
+   in the previous example, :g:`succ t` is reduced to :g:`S t`.
 
-   :opt:`Debug` ``"Cbv"`` makes :tacn:`cbv` (and its derivative :tacn:`compute`) print
-   information about the constants it encounters and the unfolding decisions it
-   makes.
+   :opt:`Debug` ``"RAKAM"`` makes :tacn:`cbn` print various debugging information.
+   ``RAKAM`` is the Refolding Algebraic Krivine Abstract Machine.
 
-.. tacn:: red
-   :name: red
+.. tacn:: hnf @simple_occurrences
 
-   This tactic applies to a goal that has the form::
+   Replaces the current goal with its
+   weak-head normal form according to the βδιζ-reduction rules, i.e. it
+   reduces the :term:`head` of the goal until it becomes a product or an
+   irreducible term. All inner βι-redexes are also reduced.  Unlike :tacn:`simpl`
+   and :tacn:`cbn`, :tacn:`hnf` does not recurse into subterms.
```

Comment:
In answer to Jason's [comment](https://github.com/coq/coq/pull/13707#discussion_r567297220), you may want to tweak this to something like "This tactic behaves similarly to `simpl` and `cbn` but unlike them, it does not recurse into subterms."